pkg/devkit/app/cluster.go

```diff
@@ -4,7 +4,165 @@
 
 package app
 
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"regexp"
+)
+
 type ClusterTemplate struct {
-	Raw   string
-	Value string `json:"value"`
+	Raw string
+}
+
+func replaceTemplateExpression(s string) string {
+	var re *regexp.Regexp
+	re = regexp.MustCompile(`"{{`)
+	s = re.ReplaceAllString(s, `{{ getv `)
+	re = regexp.MustCompile(`}}"`)
+	s = re.ReplaceAllString(s, `}}`)
+	return s
+}
+
+func getv(input interface{}) interface{} {
+	switch i := input.(type) {
+	case string:
+		return template.HTML(fmt.Sprintf(`"%s"`, i))
+	default:
+		return i
+	}
+}
+
+func (c *ClusterTemplate) Render(input ClusterConfig) (cluster *Cluster, err error) {
+	cluster = &Cluster{}
+	var tmpl *template.Template
+	raw := replaceTemplateExpression(c.Raw)
+	tmpl, err = template.New("render_cluster_config").Funcs(template.FuncMap{
+		"getv": getv,
+	}).Parse(raw)
+	if err != nil {
+		return
+	}
+	b := bytes.NewBuffer([]byte{})
+	err = tmpl.Execute(b, input)
+	if err != nil {
+		return
+	}
+	cluster.RenderJson = b.String()
+	err = json.Unmarshal(b.Bytes(), cluster)
+	return
+}
+
+type Cluster struct {
```

Comment:
There is a cluster_json_tmpl.go in models, can it be used here?

How about we use cluster.tmpl instead of cluster.json.tmpl?